src/saml.ts

```diff
@@ -38,80 +36,15 @@ import {
   validateXmlSignatureForCert,
   xpath,
 } from "./xml";
-import { certToPEM, generateUniqueId, keyToPEM, removeCertPEMHeaderAndFooter } from "./crypto";
+import { certToPEM, generateUniqueId } from "./crypto";
 import { dateStringToTimestamp, generateInstant } from "./datetime";
+import { getAdditionalParams, requestToUrl } from "./saml/common";
+import { generateServiceProviderMetadata } from "./saml/metadata";
+import { processValidlySignedPostRequest, processValidlySignedSamlLogout } from "./saml/logout";
 
 const inflateRawAsync = util.promisify(zlib.inflateRaw);
 const deflateRawAsync = util.promisify(zlib.deflateRaw);
 
-interface NameID {
```

Comment:
I believe this is a type that `passport-saml` depends on. Did you also try loading this in as a dependency of `passport-saml` and seeing if `passport-saml` still passes all of its tests?

If you need help with that LMK.

Reply:
I'll check that, thanks

Reply:
The compilation fail, but only due to an unrelated issue: the call to `getLogoutResponseUrl` is missing the `success` parameter that was added recently.

Otherwise things seem fine, tests pass.

Reply:
That seems like it shouldn't happen. I've just landed some new code in to `passport-saml`, so maybe that was the source of the mismatch? When I get some time, I'll pull this branch and verify myself for a second set of eyes.

Reply:
Ah yes, I might have not rebased on very latest commit indeed. I'll retry on the split PRs